src/MarginModels/PointSetMarginModel.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2020, Battelle Energy Alliance, LLC
+# ALL RIGHTS RESERVED
+"""
+Created on May 13 2021
+
+@author: mandd,wangc
+"""
+
+#External Modules------------------------------------------------------------------------------------
+import abc
+import sys
+import os
+import numpy as np
+import pandas as pd
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from utils import mathUtils as utils
+from utils import InputData, InputTypes
+from .MarginBase import MarginBase
+from scipy.stats import wasserstein_distance
+from sklearn.metrics import pairwise_distances
+#Internal Modules End--------------------------------------------------------------------------------
+
+
+class PointSetMarginModel(MarginBase):
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Collects input specifications for this class.
+      @ In, cls, class instance
+      @ Out, inputSpecs, InputData, specs
+    """
+    inputSpecs = super(PointSetMarginModel, cls).getInputSpecification()
+    inputSpecs.description = """ PointSet Margin Model """
+    inputSpecs.addSub(InputData.parameterInputFactory('failedDataFile',    contentType=InputTypes.InterpretedListType, descr='failed data file'))
+
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    MarginBase.__init__(self)
+
+    self.failedDataFileID = None
+    self.columnID = None
+    self.actualDataID = None
+
+
+  def _localHandleInput(self, paramInput):
+    """
+      Function to read the portion of the parsed xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, paramInput, InputData.ParameterInput, the parsed xml input
+      @ Out, None
+    """
+    MarginBase._localHandleInput(self, paramInput)
+
+    for child in paramInput.subparts:
+      if child.getName().lower() == 'failedDataFileID':
+        self.failedDataFileID = child.value
+      if child.getName().lower() == 'columnID':
+        self.columnID = child.value
+      if child.getName().lower() == 'actualDataID':
```

Comment:
remove all above .lower(), since you are comparing with the exact string, otherwise change the RHS string to lower case

Reply:
From the test, I see both columnID and actualDataID can be treated as raven variable, in this case, you need to update the _variableDict.